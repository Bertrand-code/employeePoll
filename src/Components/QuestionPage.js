import React, { useEffect } from "react";
import { useState } from "react";
import { Link, useParams } from "react-router-dom";
import { getQuestion } from "../db";
import { answerQuestion } from "../Store/questionReducer";
import { connect } from "react-redux";

function QuestionPage({ answerQuestion, answers }) {
  const { question_id } = useParams();
  const [choice, setChoice] = useState(answers[question_id]);
  const [question, setQuestion] = useState(null);
  useEffect(() => {
    getQuestion(question_id).then((question) => {
      setQuestion(question);
    });
  }, [question_id]);
  const handleSubmit = (e) => {
    e.preventDefault();
    answerQuestion(question_id, choice);
  };

  const handleChange = (e) => {
    setChoice(e.target.value);
  };
  return (
    
    question && (
      <div className="question">
        <form onSubmit={handleSubmit}>
          <input
            type="radio"
            name="options"
            id="optionOne"
            value={"optionOne"}
            onChange={handleChange}
            checked={choice === "optionOne"}
          />
          <label htmlFor="optionOne">{question?.optionOne?.text}</label>
          <br />
          <input
            type="radio"
            name="options"
            id="optionTwo"
            value={"optionTwo"}
            onChange={handleChange}
            checked={choice === "optionTwo"}
          />
          <label htmlFor="optionTwo">{question?.optionTwo?.text}</label> <br />
          <button className="btnv">Vote</button>
        </form>
        <Link to="/">
          <button className="btnb">Back</button>
        </Link>
      </div>
    )
  );
}
const mapStateToProps = (state) => {
  return {
    answers: state.questionsStore.answers,
  };
};
const mapDispatchToProps = {
  answerQuestion,
};
export default connect(mapStateToProps, mapDispatchToProps)(QuestionPage);
