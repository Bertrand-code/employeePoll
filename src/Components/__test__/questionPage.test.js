import { render, cleanup, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import { store } from "../../Store";
import QuestionPage from "../QuestionPage";

afterEach(() => {
  cleanup();
});
// const mockStore = createStore();

const TestQuestion = () => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <Routes>
          <Route path="/" element={<QuestionPage />} />
        </Routes>
      </Provider>
    </BrowserRouter>
  );
};
describe("Question", () => {
  it("should render and match snapshot", () => {
    const { container } = render(<TestQuestion />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
