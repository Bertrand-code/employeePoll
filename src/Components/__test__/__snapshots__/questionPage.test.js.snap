// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should render and match snapshot 1`] = `<div />`;
